source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -89,29 +97,47 @@
     $('#generate-evaluation-button').on('click', function () {
         var model = viewmodel.toJs();
         var modelFiltered = { Employees: _.filter(model.Employees, function (e) { return e.ResponsibleId && !e.Period; }) };
-        $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
-            type: "POST",
-            dataType: 'json',
-            contentType: 'application/json; charset=utf-8',
-            data: JSON.stringify(modelFiltered),
-            complete: function (response) {
-                var modalContainer = $('#evaluations-generated-confirm');
-                var countText = (response.responseText == '1') ? "Se ha generado 1 evaluación correctamente" : "Se han generado " + response.responseText + " evaluaciones correctamente";
-                modalContainer.find('#textCount').text(countText);
-                modalContainer.modal('show');
-            },
-            error: function () {
-                alert('Fallo interno. Por favor recargue la página.');
-            }
-        });
+        if (!viewmodel.isInvalid()) {
+            $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
+                type: "POST",
+                dataType: 'json',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify(modelFiltered),
+                complete: function (response) {
+                    var modalContainer = $('#evaluations-generated-confirm');
+                    var title = "EVALUACIONES GENERADAS"
+                    modalContainer.find('#title').text(title);
+                    var countText = (response.responseText == '1') ? "Se ha generado 1 evaluación correctamente" : "Se han generado " + response.responseText + " evaluaciones correctamente";
+                    modalContainer.find('#text').text(countText);
+                    $('#button-back').hide();
+                    $('#button-confirm').show();
+                    modalContainer.modal('show');
+                },
+                error: function () {
+                    alert('Fallo interno. Por favor recargue la página.');
+                }
+            });
+        } else {
+            var modalContainer = $('#evaluations-generated-confirm');
+            var title = "EVALUACIONES NO GENERADAS"
+            modalContainer.find('#title').text(title);
+            var text = "No se puedieron generar las evaluaciones, datos incorrectos!";
+            modalContainer.find('#text').text(text);
+            $('#button-back').show();
+            $('#button-confirm').hide();
+            modalContainer.modal('show');
+        }
```

Comment:
In here you're mixing the following logic:

- UI manipulation
- AJAX interaction
- Validation (already refactored out)
- Handling AJAX response and showing the right elements

I think you should refactor some of those out so that not all the logic is in the same place.